book/fault_proofs/deploy.md

```diff
@@ -0,0 +1,107 @@
+# Deploying OP Succinct Fault Dispute Game
+
+This guide explains how to deploy the OP Succinct Fault Dispute Game contracts using the `DeployOPSuccinctDG.s.sol` script.
+
+## Overview
+
+The deployment script performs the following actions:
+1. Deploys the `DisputeGameFactory` implementation and proxy
+2. Deploys a mock SP1 verifier for testing or use a SP1 Verifier Gateway set in the environment variable `SP1_VERIFIER_GATEWAY` for production
+3. Deploys the `OPSuccinctFaultDisputeGame` implementation
+4. Configures the factory with initial bond and game implementation
+
+## Prerequisites
+
+- [Foundry](https://book.getfoundry.sh/getting-started/installation) installed
+- Access to an Ethereum node (local or network)
+- Environment variables properly configured
+
+## Configuration
+
+Create a `.env` file in the contracts directory with the following variables:
+
+```env
+# Required Environment Variables for Game Configuration
+GAME_TYPE=42                     # Unique identifier for the game type (uint32)
+MAX_CHALLENGE_DURATION=604800    # Maximum duration for challenges in seconds
+MAX_PROVE_DURATION=86400         # Maximum duration for proving in seconds
+
+# SP1 Verifier Configuration (choose one)
+USE_SP1_MOCK_VERIFIER=true       # Set to true to deploy a mock verifier for testing
+# OR
+SP1_VERIFIER_GATEWAY=0x...       # Address of the SP1 verifier gateway for production
+```
+
+## Deployment
+
+1. Install dependencies:
+   ```bash
+   forge install
+   ```
+
+2. Change directory to contracts:
+   ```bash
+   cd contracts
+   ```
+
+3. Build the contracts:
+   ```bash
+   forge build
+   ```
+
+4. Run the deployment script:
+   ```bash
+   forge script script/fp/DeployOPSuccinctDG.s.sol --broadcast --rpc-url <RPC_URL> --private-key <PRIVATE_KEY>
+   ```
+
+## Contract Parameters
+
+The deployment script sets up the following parameters:
```

Comment:
The deployment script deploys the contract with the follow parameters:

Reply:
applied with "following"